Userland/SampleCodeModule/namedPipes.c

```diff
@@ -0,0 +1,108 @@
+/* Local headers */
+#include <namedPipes.h>
+#include <kernelTypes.h>
+#include <syscalls.h>
+#include <userstdlib.h>
+
+#define pipeName "testingPipe"
+
+void readAndClosePipe(){
+    TPid p = sys_getPid();
+    printf("Reading process with pid %d created \n", p);
+    
+    printf("Process %d opening pipe with name %s \n",p, pipeName);
+    int pipeFd[2];
+
+    if(sys_openPipe(pipeName, pipeFd)<0){
+        printf("Error while trying to open pipe");
+        return;
+    }
+
+    if(sys_close(pipeFd[1]) < 0){
+        printf("Error while trying to close pipe fd");
+        return;
+    }
+
+    char buffer[100];
+
+    if(sys_read(pipeFd[0], buffer, 100) < 0){
+        printf("Error while trying to read from pipe");
+        return;
+    }
+
+    printf("Process %d read from pipe \"%s\" the content \"%s\" \n",p, pipeName, buffer);
```

Comment:
Don't forget to null-terminate your strings! :)

```suggestion
    char buffer[101];
    ssize_t count = sys_read(pipeFd[0], buffer, 100);
    if (count < 0) {
        printf("Error while trying to read from pipe");
        return;
    }
    buffer[count] = '\0';

    printf("Process %d read from pipe \"%s\" the content \"%s\" \n",p, pipeName, buffer);
```